nengo/utils/threading.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+
+import collections
+import threading
+
+
+class ThreadLocalContext(threading.local, collections.Sequence):
```

Comment:
Or maybe `ThreadLocalStack` since this is (pretty much) a stack, and doesn't have the full functionality of a Python list.